widget/label.go

```diff
@@ -59,29 +57,59 @@ func (l *Label) Bind(data binding.String) {
 	data.AddListener(l.textListener)
 }
 
-// Refresh checks if the text content should be updated then refreshes the graphical context
+// CreateRenderer is a private method to Fyne which links this widget to its renderer
+func (l *Label) CreateRenderer() fyne.WidgetRenderer {
+	l.provider = NewRichTextWithText(l.Text)
+	l.ExtendBaseWidget(l.super())
+	l.syncSegments()
+
+	return l.provider.CreateRenderer()
+}
+
+// ExtendBaseWidget is used by an extending widget to make use of BaseWidget functionality.
+func (l *Label) ExtendBaseWidget(w fyne.Widget) {
+	if w == nil {
+		w = l
+	}
+	l.BaseWidget.ExtendBaseWidget(w)
+	if l.provider != nil {
+		l.provider.ExtendBaseWidget(w)
+	}
+}
+
+// MinSize returns the size that this label should not shrink below.
+//
+// Implements: fyne.Widget
+func (l *Label) MinSize() fyne.Size {
+	if l.provider == nil {
+		l.CreateRenderer()
+	}
+
+	return l.provider.MinSize()
+}
+
+// Refresh triggers a redraw of the label.
+//
+// Implements: fyne.Widget
 func (l *Label) Refresh() {
 	if l.provider == nil { // not created until visible
 		return
 	}
-
-	if l.Text != string(l.provider.buffer) {
-		l.provider.setText(l.Text)
-	} else {
-		l.provider.updateRowBounds() // if truncate/wrap has changed
-	}
+	l.syncSegments()
+	l.provider.Refresh()
```

Comment:
maybe this is not needed, as `l.BaseWidget.Refresh()` will already refresh it, am I missing something?

Reply:
I see, in that case `l.BaseWidget.Refresh()` should be removed, currently it refreshes twice. Because the label renderer is the rich text renderer. This could be just:
```go
func (l *Label) Refresh() {
	if l.provider == nil { // not created until visible
		return
	}
	l.syncSegments()
	l.provider.Refresh()
}
```
And the tests will pass :).